### PR TITLE
Bugfix: EventLogTarget OnOverflow=Split writes always to Info level

### PR DESCRIPTION
**Pull request by UgurAldanmaz**
Fixes: #1101
Also, Unit tests refactored and improved.

### CODE REVIEW

File: tests/NLog.UnitTests/Targets/EventLogTargetTests.cs [hunk 1]
```diff
@@ -49,198 +49,283 @@ public class EventLogTargetTests : NLogTestBase
...
+        }
+
+
+
+        private void AssertMessageCountAndLogLevelForSplittedMessages(LogLevel loglevel, EventLogEntryType expectedEventLogEntryType, Layout entryTypeLayout)
+        {
+            string testMessage = string.Join("", Enumerable.Repeat("l", MaxMessageSize + 1));
+            var entries = Write(loglevel, expectedEventLogEntryType, testMessage, entryTypeLayout, EventLogTargetOverflowAction.Split).ToList();
+
+            Assert.True(entries.Count == 2, string.Format("2 evenlogs expected. But {0} exist", entries.Count));
```
Comment (304NotModified) [inline tests/NLog.UnitTests/Targets/EventLogTargetTests.cs]: I would recommend to: or use assert.areEquals, or write a helpers for this line.

Comment (304NotModified) [inline tests/NLog.UnitTests/Targets/EventLogTargetTests.cs]: I mean `.equal`.  Confused by nunit/xunit

**Comment by 304NotModified** [comment]
Looks nice! Good renames! 

One code comment in the test case,  and maybe you could improve the code coverage to test with a string with length of exactly 2*max. 

See https://codecov.io/github/NLog/NLog/commit/1c8874593b739517972fcfb5c25c172538667a2e

Partly means (yellow) that only one (outcome of the)  condition is tested (this case true) .

File: tests/NLog.UnitTests/Targets/EventLogTargetTests.cs (hunk 1 above)
Comment (UgurAldanmaz) [inline tests/NLog.UnitTests/Targets/EventLogTargetTests.cs]: Thanks! It was Equal. But Equal method has not message parameter. So I used the True method be able to use the message parameter.

Comment (304NotModified) [inline tests/NLog.UnitTests/Targets/EventLogTargetTests.cs]: Yeah I know, the philosophy of xunit is that message parameters are bad. Not fully agree with that.

Comment (304NotModified) [inline tests/NLog.UnitTests/Targets/EventLogTargetTests.cs]: https://github.com/xunit/xunit/issues/350

**Comment by UgurAldanmaz** [comment]
Thanks @304NotModified! I am going to improve code coverage by reviewing the codecov results.

File: tests/NLog.UnitTests/Targets/EventLogTargetTests.cs (hunk 1 above)
Comment (UgurAldanmaz) [inline tests/NLog.UnitTests/Targets/EventLogTargetTests.cs]: > Assertions with messages are like giving up on clear code in favor of comments, and with all the requisite danger: if you change the assert but not the message, then it leads you astray.

He seems right.

**Comment by 304NotModified** [comment]
Just let me know when we look at it :)

**Comment by UgurAldanmaz** [comment]
I did some changes for your review comments and codecov analyse. I have committed them in same day, but forgot to inform :)

Code coverage did not increase. Maybe we need to consider to write unit tests for other pieces fo EventLogTarget. I can do this by opening new issue.

Thanks for reminder!

**Comment by 304NotModified** [comment]
Well codecov is buggy? The links won't work? 

also browing manual to it wont work:
https://codecov.io/github/NLog/NLog/src/NLog/Targets/EventLogTarget.cs?ref=c6d43bc392ea434f7cd545968b66f2d4df00e962

**Comment by 304NotModified** [comment]
Thanks! It has been merged :)

**Comment by UgurAldanmaz** [comment]
I thank you!

It seems that code cov links does not works anymore (gives 404).

**Pull request merged**